pkg/cloudprovider/azure/block_store.go

```diff
@@ -211,14 +222,13 @@ func (b *blockStore) CreateSnapshot(volumeID, volumeAZ string, tags map[string]s
 	defer cancel()
 
 	_, errChan := b.snaps.CreateOrUpdate(b.resourceGroup, *snap.Name, snap, ctx.Done())
-
 	err = <-errChan
 
 	if err != nil {
 		return "", errors.WithStack(err)
 	}
 
-	return snapshotName, nil
+	return fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/snapshots/%s", b.subscription, b.resourceGroup, snapshotName), nil
```

Comment:
@skriss do you think it would be worth creating `getComputeResourceName(subscription, resourceGroup, resource, name string) string`? Then we could call it here (with resource=snapshots) and wherever we call `getFullDiskName` (with resource=disks).

Reply:
good idea. the fewer hardcoded URI templates the better.